File: javascript/raindrops/raindrops.js
const factors = [
  [3, 'Pling'],
  [5, 'Plang'],
  [7, 'Plong'],
]

module.exports = class Raindrops {
  convert(n) {
    const drops = factors.reduce((ds, [f, word]) => n % f === 0 ? ds + word : ds, '')
    return drops || String(n)
  }
}

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

type Pair struct {
	Factor int
	Word   string
}

func Convert(n int) string {
	pairs := []Pair{
		Pair{3, "Pling"},
		Pair{5, "Plang"},
		Pair{7, "Plong"},
	}

	s := ""
	for _, p := range pairs {
		if n%p.Factor == 0 {
			s += p.Word
		}
	}

	if s == "" {
		return strconv.Itoa(n)
	}
	return s
}
